restapi/tags.go

```diff
@@ -0,0 +1,24 @@
+package restapi
+
+import (
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/uwblueprint/shoe-project/internal/database/models"
+	"github.com/uwblueprint/shoe-project/restapi/rest"
+)
+
+func (api api) ReturnAllUniqueTags(w http.ResponseWriter, r *http.Request) render.Renderer {
+	var tags []models.Tag
+	err := api.database.Distinct("name").Find(&tags).Error
+	if err != nil {
+		return rest.ErrInternal(api.logger, err)
+	}
+
+	var length = len(tags)
+	names := make([]string, length)
+	for i, t := range tags {
+		names[i] = t.Name
```

Comment:
Aside: holy we really need generics for `map`, `filter` and `reduce` in this language.